app/src/main/java/com/example/scotlandyard/lobby/ClientLobby.java

```diff
@@ -4,202 +4,137 @@
-                } catch (InterruptedException e) {
-                    Log.e("InterruptedExeption", e.getMessage(), e.getCause());
-                }
-
-                runOnUiThread(new Runnable() {
-                    @Override
-                    public void run() {
-                        ((TextView) findViewById(R.id.textConnectionInfo)).setText("Waiting for server to start the game");
-                        ((ProgressBar) findViewById(R.id.progressBarConnection)).setProgress(80);
-                    }
-                });
-
-                clientService.send(new Message("GET_GAME_DATA"));
-            }
-        }).start();
+    public void showConnected(String endpointName) {
+        Log.d(logTag, "should not happen");
     }
 
     @Override
-    public void onGameData(Game game) {
-        Log.d(logTag, "onGameData() : Name = " + game.getGameName() + "; #Players = " + game.getPlayers());
-
-        Game gameReceived = (Game) game;
+    public void startGame(Game game) {
+        Log.d(logTag, "Game start initiated by server!");
 
-        this.game.getPlayers().clear();
+        ((TextView) findViewById(R.id.textConnectionInfo)).setText("Das Spiel wird gestartet...");
+        ((ProgressBar) findViewById(R.id.progressBarConnection)).setProgress(100);
 
-        this.game.getPlayers().addAll(gameReceived.getPlayers());
-
-        ((ArrayAdapter) connectedPlayersListAdapter).notifyDataSetChanged();
+        Intent gameStartIntent = new Intent(ClientLobby.this, GameMap.class);
+        gameStartIntent.putExtra("USERNAME", player.getNickname());
+        gameStartIntent.putExtra("IS_SERVER", false);
+        startActivity(gameStartIntent);
     }
 
     @Override
-    public void onMessage(Message message) {
-        Log.d(logTag, "onMessage() : Message = " + message.toString());
-
-        String receivedMessage = ((Message) message).getMessage();
-
-        if (receivedMessage.equals("START_GAME")) {
-            Log.d(logTag, "Game start initiated by server!");
-            ((ProgressBar) findViewById(R.id.progressBarConnection)).setProgress(100);
-
-            Intent gameStartIntent = new Intent(ClientLobby.this, GameMap.class);
-            gameStartIntent.putExtra("USERNAME", username);
-            gameStartIntent.putExtra("IS_SERVER", false);
-            startActivity(gameStartIntent);
-        }
+    public void showConnectionFailed(String endpointName) {
+        Log.d(logTag, "should not happen");
     }
 
     @Override
-    public void onSendMove(SendMove sendMove) {
-        Log.d(logTag, "onSendMove() : Message = " + sendMove.toString());
+    public void showDisconnected(String endpointName) {
+        Log.d(logTag, "onDisconnected() : " + endpointName);
 
-    }
-
-    @Override
-    public void onRoadMapEntry(Entry entry) {
+        String notification = "Verbindung zum Server verloren.";
+        Toast.makeText(getApplicationContext(), notification, Toast.LENGTH_LONG).show();
 
```

Comment:
Wenn vom Server disconnected, zu welcher activity sollte man kommen? In der lobby (ohne server bleiben)?!